### PR TITLE
Enhanced fire alarm

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

- New fire alarm sprite
- Area fire alarms can be muted
- Adds alert level to examine text
- Adds balloon alerts

## Why It's Good For The Game

You can mute the area when it's been sounding off for 20 minutes and no atmos tech in sight. Refreshed sprite.

https://user-images.githubusercontent.com/83487515/224538329-a1179a29-dbcc-431c-bccb-a54d52071d21.mp4

## Changelog

:cl: LT3
add: Enhanced fire alarm, now with a mute button
qol: You can acknowledge the fire alert and mute the sound from the panel
/:cl:

### CODE REVIEW

**Comment by san7890** [comment]
image

did you forget the icons for this state?

**Comment by VladinXXV** [comment]
the video has no sound to begin with so it doesnt prove much to anyone

**Comment by VladinXXV** [comment]
> we don't have these

yet. you're missing the big picture melbert...

**Comment by lessthnthree** [comment]
> i assume these are alert levels? you don't have to bring these over - we don't have these image

We have those colors downstream, and would allow us to delete our overrides and use the base TG code. Also if you decide to add colors in the future, it's available.

I can remove them though if it's a big deal.

**Comment by lessthnthree** [comment]
> Maybe we can delegate generating the fire alarm overlay to `SSsecurity_level` instead then? Even cache it since we're in the neighborhood? That way you can specify different icon file sources for the overlay

I'm okay with making little change as possible to it at the moment, so I'll remove the additional levels.

SSsecurity_level could use some additional things, not just fire alarms and has been on my radar, but it'd be a larger thing.

**Comment by lessthnthree** [comment]
> I think the only case where a fire alarm should get muted is if they're off. Don't like the sound? Get the atmospheric technicians to clear the alarm.

This is real life functionality on fire alarms here in multiple unit dwellings and it's fantastic QoL feature.

The point of a fire alarm is to notify (and wake up) the occupants of the space. Once the alarm has alerted everyone and they have acknowledged this by pushing the button, there is no further reason to continue the sound. 
A lower tech scenario more common, do you push the silence button when you burn food in the kitchen and the smoke detector goes off? Or do you let and sit it run after everyone is aware that it's alerting to something.

When the fire alarm goes off in a commercial space, do they leave it ringing or silence it and leave only the flashing lights? Leaving the audio on longer than it needs to be is just an annoyance.

**Comment by Zergspower** [comment]
Yeah alarms can be off and still flash - it's not that big of a stretch to have the same function we have in current time inside a future spess game.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I don't think you should be able to mute fire locks, I think they add a lot to the ambience of a crumbling infrastructure, and the game would feel significantly worse for me if people did this.

**Comment by Mothblocks** [comment]
> When the fire alarm goes off in a commercial space, do they leave it ringing or silence it and leave only the flashing lights? Leaving the audio on longer than it needs to be is just an annoyance.

I had a fire alarm go off in a building I was in about 3 weeks ago and the alarm kept going until the alarms were off as well

**Comment by VladinXXV** [comment]
> I don't think you should be able to mute fire locks, I think they add a lot to the ambience of a crumbling infrastructure.

There is a significant difference between "crumbling infrastructure" and "someone opened an airlock for 2 seconds on tramstation and now every air alarm in a 100 tile radius is crying". The ability to mute firelocks in itself doesn't make for a loss to this ambience because in a real emergency _no one is going to waste their time with it_. It serves as a sanity-saving measure for the many times firelocks are a nuisance for no meaningful reason.

**Comment by Mothblocks** [comment]
> someone opened an airlock for 2 seconds on tramstation and now every air alarm in a 100 tile radius is crying

> It serves as a sanity-saving measure for the many times firelocks are a nuisance for no meaningful reason.

As usual with fire lock PRs, we are bandaiding over that fire locks going off when they shouldn't is the real problem. It's been a year+ of people trying to convince me things like this are fine and I am still no closer to being convinced.

**Comment by VladinXXV** [comment]
> As usual with fire lock PRs, we are bandaiding over that fire locks going off when they shouldn't is the real problem. It's been a year+ of people trying to convince me things like this are fine and I am still no closer to being convinced.

Let me try this, then: Can we have an option to mute firelocks/fire alarms _specifically_ in the Sound settings? Your precious ambience means little to me; my ears hurt and I want a solution now, not in 30 years when someone makes firelocks work properly. This would be clientside and I don't see why you should care if I'm "damaging" my own ambience. We do it for several other sounds in the game (including the ambience sounds themselves).

**Comment by Mothblocks** [comment]
I think we can get away with making them quieter in general, I don't think a preference is necessary

**Comment by VladinXXV** [comment]
> I think we can get away with making them quieter in general, I don't think a preference is necessary

I appreciate what I assume is an attempt to find compromise, but making them quieter does not necessarily improve the situation. To a certain point, it can even make it worse - it can be quite annoying to be barely conscious of a sound for so long. No matter the volume or pitch, the main issue here is the repetitive and continuous nature of the sound. Legitimate cause or not, oftentime the fire alarms do not stop once they have started. Engineering is either lazy, incompetent, has more pressing matters to deal with, or does not exist. I think the preference is a much solider solution, which is futureproofed against however grating the sound may become in the future - those who would complain can be told "just turn it off".

**Comment by lessthnthree** [comment]
I'd like to work on firelocks so they have less false positives firing, but even in the perfect system I think there should still be a mute function, which again exists in the real world as a QoL thing, and yes there will be times engineering can't or won't get to you for whatever reason.

**Comment by Mothblocks** [comment]
>but even in the perfect system I think there should still be a mute function

I don't agree sorry

**Comment by CRITAWAKETS** [comment]
should probably mention where the sprites come from for courtesy

File: code/game/machinery/firealarm.dm [hunk 1]
```diff
@@ -168,8 +172,7 @@
 	obj_flags |= EMAGGED
 	update_appearance()
 	if(user)
-		user.visible_message(span_warning("Sparks fly out of [src]!"),
-							span_notice("You override [src], disabling the speaker."))
+		user.balloon_alert(user, "speaker disabled!")
```
Comment (ZephyrTFA) [inline code/game/machinery/firealarm.dm:175]: this makes it so that emagging is no longer obvious to observers

**Comment by Ryusenshu** [comment]
i honestly dont like anything in this pr
Not the sprites, not the mute function or anything really

Pixelhunting is already annoying enough, doesnt have to be even smaller

**Comment by Mothblocks** [comment]
Hrm. Yeah I'm also not a fan of the sprites being so much smaller. Legitimately sad balance change, I like people running through rooms and quickly smashing the fire alarm to stop people in their tracks, it's good environmental combat. That being said I think we can replicate that in a future PR by making it so clicking on a wall with an empty hand clicks on the wallframe (or equivalent) on top of it, except/prohibitng false walls. That way we can fix the pixel hunt issue without it causing problems for spriters.

**Comment by Mothblocks** [comment]
(To be unambiguous, don't worry about the sprites unless you want to personally change them, you're causing a problem that we *should* be fixing separately)

**Comment by lessthnthree** [comment]
Not worth respriting in the same size as the old alarm if it's just a cosmetic change.

File: code/game/machinery/firealarm.dm (hunk 1 above)
Comment (lessthnthree) [reply code/game/machinery/firealarm.dm:175]: okay, I'll put it back

**Comment by lessthnthree** [comment]
Based on that, I did a new sprite that's the same size as the old fire alarm:

https://user-images.githubusercontent.com/83487515/224538388-4eb17ab0-f7a1-4240-91c3-9bb52df46004.mp4

**Comment by ZephyrTFA** [comment]
I don't really think that sprite is an improvement

**Comment by Ryusenshu** [comment]
> I don't really think that sprite is an improvement

It IS better than the one that was before this in the pr
Thats what my "i like" was based on

There are still issues to it, about visibility in low light as an example, or the different states

**Comment by lessthnthree** [comment]
It's not the final version of sprite, thus the draft. I did want feedback if it was heading in the right direction and what flaws you see, such as low light as mentioned.

**Comment by Ryusenshu** [comment]
> It's not the final version of sprite, thus the draft. I did want feedback if it was heading in the right direction for what you want to see on the alarm, and what flaws you see, such as low light as mentioned.

For the enabled fire alarm, id like to see the spinning circle in green with a red flashing background in intervals maybe
And for the states i mentioned, you used to be able to see if sensors were off (multitool or silicon ctrl-click), so it might need an added lamp somewhere (maybe near the top right?)

**Comment by lessthnthree** [comment]
> > It's not the final version of sprite, thus the draft. I did want feedback if it was heading in the right direction for what you want to see on the alarm, and what flaws you see, such as low light as mentioned.
> 
> For the enabled fire alarm, id like to see the spinning circle in green with a red background flashing in intervals That would make low light a tad better
> 
> And for the states i mentioned, you used to be able to see if sensors were off (multitool or silicon ctrl-click), so it might need an added lamp somewhere (maybe near the top right?)
> 
> So yeah i like the sprite but its not on full parity yet

I was thinking of making the green light go out if the sensors are disabled, so it's apparent. An emagged one would flash a blue error screen. I haven't added any light overlays yet so that should help make them stand out.

**Comment by dessysalta** [comment]
I think the mute should be tied to access (say, maints or something?), but definitely not removed. My first thought was that one tragedy where 3 people died after silencing an alarm for ammonia detectors at a hockey rink. Alarms can be silenced in real life, and just like real life, silencing them doesn't prevent stupidity.

**Comment by Mothblocks** [comment]
I don't want a mute because I think it's important for the ambience and I would deny this PR if it kept it

**Comment by lessthnthree** [comment]
> I don't want a mute because I think it's important for the ambience and I would deny this PR if it kept it

I'll close this then, and if anything it'll come around sometime as just a sprite in that case. I can take the spritework to Discord.

**Comment by bluetshirtguy** [comment]
It's ass ambiance and ear agony but whatever you think I guess.

**Comment by Mothblocks** [comment]
It's good ambience, might be a bit loud but can be touched down

**Comment by dessysalta** [comment]
It would be good ambience if it didn't come up so often with so many opportunities to counter the issues often associated with it. You hear it nearly every shift at some point, and most of the time it just means "use maintenance" or "fix the hole/temp/etc". Ambience has to be used sparingly and in the right circumstances, lest you get accustomed to it and stop feeling concerned, but I'm probably not well versed enough in sound design and gameplay loops to think of an accurate solution (which is already hard because /TG/'s whole loop is pretty much nonstop chaos).

I do think there's a compromise we haven't reached, though. As mentioned before, tying the mute to access (like engi or maints) could be an adequate solution that leaves the ambience in play while still allowing exceptions if players have grown tired of it. As it stands the only way to silence it is to emag it (which doesn't work most of the time) or destroy the fire alarms/locks entirely.

**Comment by Mothblocks** [comment]
image

**Pull request closed without merging**